tests/sample_size/scripts/test_multiple_testing.py

```diff
@@ -0,0 +1,97 @@
+import unittest
+from unittest.mock import patch
+
+from sample_size.sample_size_calculator import SampleSizeCalculator
+
+
+class MultipleTestingTestCase(unittest.TestCase):
+    def setUp(self):
+        self.test_lower = 100
+        self.test_upper = 1000
+        self.test_metric_type = "boolean"
+        self.test_probability = 0.05
+        self.test_mde = 0.02
+        self.test_metric_metadata = {"probability": self.test_probability, "mde": self.test_mde}
+        self.test_metric = {"metric_type": self.test_metric_type, "metric_metadata": self.test_metric_metadata}
+
+    @patch(
+        "sample_size.sample_size_calculator.SampleSizeCalculator._get_single_sample_size",
+        side_effect=[100, 100, 1000, 1000],
+    )
+    @patch("sample_size.multiple_testing.MultipleTestingMixin._expected_average_power", side_effect=[0.5, 0.8])
+    def test_get_multiple_sample_size_too_low(self, mock_expected_average_power, mock_get_single_sample_size):
+        calculator = SampleSizeCalculator()
+        calculator.register_metrics([self.test_metric, self.test_metric])
+        sample_size = calculator.get_sample_size()
+        self.assertEqual(mock_get_single_sample_size.call_count, 4)
+        self.assertEqual(mock_expected_average_power.call_count, 2)
```

Comment:
Could help clarify why we're asserting 2 and 4 here? It's not clear to me why these are _supposed_ to be 2 and 4.

Reply:
Every time we run multiple testing calculator, we are calculating each metric's required sample size and get the lower and upper bound. Thus, `mock_get_single_sample_size` will be called for `len(metrics)*2` times. In this test case we have 2 recursions so `mock_expected_average_power` is expected to be called twice.

Reply:
I see. This dovetails with my comment below. Perhaps we could parameterize the number of recursive calls?

```python
    @paramterized.expand(product([0, 1], [1, 2, 3]))
    @patch(
        "sample_size.sample_size_calculator.SampleSizeCalculator._get_single_sample_size",
        side_effect=[100, 100, 1000, 1000],
    )
    @patch("sample_size.multiple_testing.MultipleTestingMixin._expected_average_power")
    def test_get_multiple_sample_size_quickly_converge(self, mock_expected_average_power, mock_get_single_sample_size, power_guess, n_recursive_calls):
        mock_expected_average_power.side_effect = ([power_guess] * n_recursive_calls) + [0.8]
        calculator = SampleSizeCalculator()
        calculator.register_metrics([self.test_metric, self.test_metric])
        sample_size = calculator.get_sample_size()
        self.assertEqual(mock_get_single_sample_size.call_count, 2 * n_recursive_calls)
        self.assertEqual(mock_expected_average_power.call_count, n_recursive_calls)
        self.assertEqual(sample_size, ((self.test_upper + self.test_lower) / 2 + self.test_lower) / 2)
```

Reply:
^whoops, logged into wrong github account